probe/main.go

```diff
@@ -57,7 +58,10 @@ func main() {
 			log.Printf("exposing Prometheus endpoint at %s%s", *httpListen, *prometheusEndpoint)
 			http.Handle(*prometheusEndpoint, makePrometheusHandler())
 		}
-		go func(err error) { log.Print(err) }(http.ListenAndServe(*httpListen, nil))
+		go func() {
+			err := http.ListenAndServe(*httpListen, nil)
+			log.Print(err)
+		}()
```

Comment:
"The function value and parameters are evaluated as usual in the calling goroutine" - https://golang.org/ref/spec#Go_statements

ListenAndServe blocks, so go f(ListenAndServe()) also blocks.